BoxContentSDK/BoxContentSDK/Categories/UIApplication+ExtensionSafeCompilation.m

```diff
@@ -0,0 +1,47 @@
+//
+//  UIApplication+ExtensionSafeCompilation.m
+//  app
+//
+//  Created by Scott Liu on 5/13/16.
+//  Copyright © 2016 Box. All rights reserved.
+//
+
+#import "UIApplication+ExtensionSafeCompilation.h"
+
+@implementation UIApplication (ExtensionSafeCompilation)
+
++ (UIApplication *)box_sharedApplication
+{
+#ifdef EXTENSION
+    // If we are compiling from extensions, return nil here since UIApplication isn't extension-safe.
+    return nil;
+#else
+    // If we are compiling from the main app, return normal sharedApplication.
+    return [UIApplication sharedApplication];
+#endif
+}
+
+- (BOOL)box_canOpenURL:(NSURL *)url
+{
+#ifdef EXTENSION
```

Comment:
this should be to specify that the current target is an extension. What about TARGET_IS_EXTENSION ?